src/reducers/versions.tsx

```diff
@@ -594,6 +614,56 @@ type FetchDiffParams = {
   path?: string;
 };
 
+type GetCompareInfoKeyParams = {
+  addonId: number;
+  baseVersionId: number;
+  headVersionId: number;
+  path?: string;
+};
+
+export const getCompareInfoKey = ({
+  addonId,
+  baseVersionId,
+  headVersionId,
+  path,
+}: GetCompareInfoKeyParams) => {
+  return [addonId, baseVersionId, headVersionId, path].join('/');
+};
+
+export const getCompareInfo = (
+  versions: VersionsState,
+  addonId: number,
+  baseVersionId: number,
+  headVersionId: number,
+  path?: string,
+) => {
```

Comment:
I'm nervous about TS inferring the `undefined` part of this return type. Is that fixed now? Or do we need to provide an explicit return type for it?

Reply:
Yes, it was fixed for another function so I guess it is the same here.

Reply:
If I change the prop type in `Compare`, I get a TS error because `undefined` is not handled, so I guess we're good.